File: hermes/src/main/java/org/eclipse/paho/android/service/contract/SubscribeActionListener.java
package org.eclipse.paho.android.service.contract;

import org.eclipse.paho.android.service.event.MqttAction;

/**
 * 订阅/解订阅相关操作回调
 * @author zhouL
 * @date 2023/9/27
 */
public interface SubscribeActionListener {

    void onSuccess(MqttAction action, String clientKey, String[] topic);

    void onFailure(MqttAction action, String clientKey, String[] topic, String errorMsg, Throwable t);
}

File: hermes/src/main/java/com/lodz/android/hermes/mqtt/base/contract/SubscribeActionListener.kt
package com.lodz.android.hermes.mqtt.base.contract

import com.lodz.android.hermes.mqtt.base.bean.eun.MqttAction

/**
 * 订阅操作回调
 * @author zhouL
 * @date 2023/10/12
 */
interface SubscribeActionListener {

    /** 主键[clientKey]的订阅操作[action]成功，主题[topics] */
    fun onSuccess(action: MqttAction, clientKey: String, topics: Array<String>)

    /** 主键[clientKey]的订阅操作[action]失败，主题[topics]，提示语[errorMsg]，异常[t] */
    fun onFailure(action: MqttAction, clientKey: String, topics: Array<String>, errorMsg: String, t: Throwable)

}